client/src/components/viewers/controls/FileControlsDropdownPopover.vue

```diff
@@ -0,0 +1,101 @@
+<!-- Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS -->
+
+<template>
+  <ion-list class="file-controls-dropdown-popover">
+    <div
+      v-if="parentHistory.length > 0"
+      @click="onBackClicked"
+      class="back-button"
+    >
+      <ion-icon :icon="chevronBack" />
+      <ion-text class="title">
+        {{ parentNames.slice(-1)[0] }}
+      </ion-text>
+    </div>
+    <div
+      class="divider"
+      v-if="parentHistory.length > 0"
+    />
+    <file-controls-dropdown-item
+      v-for="item in displayedItems.items"
+      :key="item.id"
+      :is-active="item.id === displayedItems.activeItem"
+      :item="item"
+      @click="onOptionClick"
+    />
+  </ion-list>
+</template>
+
+<script setup lang="ts">
+import { FileControlsDropdownItem, FileControlsDropdownItemContent, FileControlsDropdownGroup } from '@/components/viewers';
+import { IonIcon, IonList, IonText, popoverController } from '@ionic/vue';
+import { ref } from 'vue';
+import { chevronBack } from 'ionicons/icons';
+import { Translatable } from 'megashark-lib';
+
+const props = defineProps<{
+  items: FileControlsDropdownGroup;
+}>();
+
+const displayedItems = ref<FileControlsDropdownGroup>(props.items);
+const parentHistory = ref<Array<FileControlsDropdownGroup>>([]);
```

Comment:
Since those variables seem to be synchronous (you always do the same operation on both at once, unless I missed something) I would group them together:
`const blabla: Array<{history: FileControlsDropdownGroup, names: Translatable}> = [];`
This ensures that they stay synced.